url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/net/http/PieceInfoList.java

```diff
@@ -0,0 +1,91 @@
+package net.hetimatan.net.http;
+
+import java.util.Collections;
+import java.util.LinkedList;
+
+public class PieceInfoList {
+	private LinkedList<PieceInfo> mInfo = new LinkedList<PieceInfo>();
+
+	public int size() {
+		return mInfo.size();
+	}
+
+	public PieceInfo getPieceInfo(int index) {
+		return mInfo.get(index);
+	}
+
+	public void append(long start, long end) {
+		mInfo.add(new PieceInfo(start, end));
+		normalize();
+	}
+
+	// #patternA
+	// <-----><----->
+	//    <------>
+	//    ↑start　↑end
+	//
+	// #patternB
+	// <-----><----->
+	//   <->
+	//
+	//
+	// #patternC
+	//  <-----><----->
+	// <-------->
+	//
+	//
+	public void remove(long start, long end) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.